helm-charts/visualqna/templates/nginx-deployment.yaml

```diff
@@ -0,0 +1,94 @@
+apiVersion: v1
+data:
+  default.conf: |+
+    # Copyright (C) 2024 Intel Corporation
+    # SPDX-License-Identifier: Apache-2.0
+
+
+    server {
+        listen       80;
+        listen  [::]:80;
+
+        location /home {
+            alias  /usr/share/nginx/html/index.html;
+        }
+
+        location / {
+            proxy_pass http://{{ include "ui.fullname" (index .Subcharts "visualqna-ui") }}:{{ index .Values "visualqna-ui" "service" "port" }};
+            proxy_set_header Host $host;
+            proxy_set_header X-Real-IP $remote_addr;
+            proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
+            proxy_set_header X-Forwarded-Proto $scheme;
+        }
+
+        location /v1/visualqna {
+            proxy_pass http://{{ include "visualqna.fullname" . }}:{{ .Values.service.port }};
+            proxy_set_header Host $host;
+            proxy_set_header X-Real-IP $remote_addr;
+            proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
+            proxy_set_header X-Forwarded-Proto $scheme;
+        }
+
+    }
+
+kind: ConfigMap
+metadata:
+  name: {{ include "visualqna.fullname" . }}-nginx-config
+---
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: {{ include "visualqna.fullname" . }}-nginx
+  labels:
+    {{- include "visualqna.labels" . | nindent 4 }}
+    app: {{ include "visualqna.fullname" . }}-nginx
+spec:
+  selector:
+    matchLabels:
+      {{- include "visualqna.selectorLabels" . | nindent 6 }}
+      app: {{ include "visualqna.fullname" . }}-nginx
+  template:
+    metadata:
+      labels:
+        {{- include "visualqna.selectorLabels" . | nindent 8 }}
+        app: {{ include "visualqna.fullname" . }}-nginx
+    spec:
+      containers:
+      - image: nginx:1.27.1
+        imagePullPolicy: {{ .Values.image.pullPolicy }}
+        name: nginx
+        volumeMounts:
+        - mountPath: /etc/nginx/conf.d
+          name: nginx-config-volume
+      securityContext: {}
+      volumes:
+      - configMap:
+          defaultMode: 420
+          name: {{ include "visualqna.fullname" . }}-nginx-config
+        name: nginx-config-volume
+      {{- with .Values.nodeSelector }}
+      nodeSelector:
+        {{- toYaml . | nindent 8 }}
+      {{- end }}
+      {{- with .Values.affinity }}
+      affinity:
+        {{- toYaml . | nindent 8 }}
+      {{- end }}
+      {{- with .Values.tolerations }}
+      tolerations:
+        {{- toYaml . | nindent 8 }}
+      {{- end }}
+---
+apiVersion: v1
+kind: Service
+metadata:
+  name: {{ include "visualqna.fullname" . }}-nginx
+spec:
+  ports:
+  - port: 80
+    protocol: TCP
+    targetPort: 80
+  selector:
+    {{- include "visualqna.selectorLabels" . | nindent 4 }}
+    app: {{ include "visualqna.fullname" . }}-nginx
+  type: NodePort
```

Comment:
We may need to make it configurable like what we do in PR #506

Reply:
Fixed